### PR TITLE
Improve the installerBuild task to produce the installers

### PR DESCRIPTION
**Pull request by ggalmazor**
Now the Gradle task will automate the generation of all the installer if the installBuilderHome variable is set.

- The installers are generated following the filename templates we require in our releases.
- The task will only work in Linux and macOS hosts.
- The installers are built and zipped by the task, and they can be found at `build/installers`

The way to run the task hasn't changed: 
`./gradlew clean build installerBuild -xtest -PwarMode=installer`

### CODE REVIEW

File: installer.gradle [hunk 1]
```diff
@@ -24,5 +26,21 @@ task installerBuild(dependsOn: [installerClean], type: Copy) {
...
     file("${buildDir}/installer/files/${archivesBaseName}-${version}.war").renameTo(file("build/installer/files/ODKAggregate.war"))
+
+    if (installBuilderHome != null) {
+      run("mkdir ${buildDir}/installers")
+      setXmlValue("${buildDir}/installer/buildWar.xml", "outputDirectory", "${buildDir}/installers")
+
+      [["linux-x64", "Linux-x64.run"],
+       ["linux", "Linux.run"],
+       ["windows", "Windows.exe"],
+       ["osx", "macOs.app"]
```
Comment (yanokwa) [inline installer.gradle]: s/macOs/macOS/

File: gradle.properties.example [hunk 2]
```diff
@@ -2,3 +2,6 @@
 #packerZip=https://releases.hashicorp.com/packer/1.3.4/packer_1.3.4_linux_amd64.zip
 #packerZip=https://releases.hashicorp.com/packer/1.3.4/packer_1.3.4_windows_amd64.zip
 #packerZip=https://releases.hashicorp.com/packer/1.3.4/packer_1.3.4_darwin_amd64.zip
+
+#Uncomment and set the path to the installBuilder installation directory
+#installBuilderHome=/opt/installbuilder-18.10.0
```
Comment (yanokwa) [inline gradle.properties.example]: I ran `./gradlew clean installerBuild --stacktrace`. Noticed that if the `installBuilderHome` path has a space in it, the command fails with: `Cannot run program ""/Applications/BitRock": error=2, No such file or directory`

 I've tried the following paths: 
```
installBuilderHome="/Applications/BitRock InstallBuilder Enterprise 19.5.0"
installBuilderHome="/Applications/BitRock\ InstallBuilder\ Enterprise\ 19.5.0
```

File: installer.gradle [hunk 3]
```diff
@@ -24,5 +26,21 @@ task installerBuild(dependsOn: [installerClean], type: Copy) {
...
+      ].forEach({ pair ->
+        setXmlValue("${buildDir}/installer/buildWar.xml", "installerFilename", "ODK-Aggregate-${version}-${pair[1]}")
+        run("${installBuilderHome}/bin/builder build ${buildDir}/installer/buildWar.xml ${pair[0]}")
+        run("zip ODK-Aggregate-${version}-${pair[1]}.zip ODK-Aggregate-${version}-${pair[1]}", "${buildDir}/installers")
```
Comment (yanokwa) [inline installer.gradle]: Using this PR, I get a very small Mac app. 
```
 35M ODK-Aggregate-v2.0.2-29-gefab50dc-Linux-x64.run.zip
 35M ODK-Aggregate-v2.0.2-29-gefab50dc-Linux.run.zip
 35M ODK-Aggregate-v2.0.2-29-gefab50dc-Windows.exe.zip
238B ODK-Aggregate-v2.0.2-29-gefab50dc-macOs.app.zip
```

When I give the buildWar.xml to InstallBuilder directly, then I get a full-sized app.
```
/usr/local/installbuilder/bin/builder build build/installer/buildWar.xml osx
```

My guess is that on macOS, an app isn't a flat-file, it's a directory. 
```
ODK-Aggregate-v2.0.2-29-gefab50dc-macOs.app
ODK-Aggregate-v2.0.2-29-gefab50dc-macOs.app/Contents
ODK-Aggregate-v2.0.2-29-gefab50dc-macOs.app/Contents/MacOS
ODK-Aggregate-v2.0.2-29-gefab50dc-macOs.app/Contents/MacOS/installbuilder.sh
ODK-Aggregate-v2.0.2-29-gefab50dc-macOs.app/Contents/MacOS/osx-intel
ODK-Aggregate-v2.0.2-29-gefab50dc-macOs.app/Contents/Resources
ODK-Aggregate-v2.0.2-29-gefab50dc-macOs.app/Contents/Resources/installbuilder
ODK-Aggregate-v2.0.2-29-gefab50dc-macOs.app/Contents/Resources/installbuilder.icns
ODK-Aggregate-v2.0.2-29-gefab50dc-macOs.app/Contents/Resources/osx-intel.dat
ODK-Aggregate-v2.0.2-29-gefab50dc-macOs.app/Contents/Info.plist
```

So you should probably add a `-r` flag to the zip.
```
run("zip -r ODK-Aggregate-${version}-${pair[1]}.zip ODK-Aggregate-${version}-${pair[1]}", "${buildDir}/installers")
```

**Comment by ggalmazor** [comment]
Hi, @yanokwa! Did you have a chance to review the last commit with the things you suggested?

**Comment by ggalmazor** [comment]
Awesome, thanks!

**Pull request merged**